roborock/devices/local_channel.py

```diff
@@ -39,18 +42,83 @@ class LocalChannel(Channel):
     format most parsing to higher-level components.
     """
 
-    def __init__(self, host: str, local_key: str):
+    def __init__(self, host: str, local_key: str, local_protocol_version: LocalProtocolVersion | None = None):
         self._host = host
         self._transport: asyncio.Transport | None = None
         self._protocol: _LocalProtocol | None = None
         self._subscribers: CallbackList[RoborockMessage] = CallbackList(_LOGGER)
         self._is_connected = False
-
-        self._decoder: Decoder = create_local_decoder(local_key)
-        self._encoder: Encoder = create_local_encoder(local_key)
+        self._local_key = local_key
+        self._local_protocol_version = local_protocol_version
+        self._connect_nonce = get_next_int(10000, 32767)
+        self._ack_nonce: int | None = None
+        self._update_encoder_decoder()
+
+    def _update_encoder_decoder(self):
+        self._encoder = create_local_encoder(
+            local_key=self._local_key, connect_nonce=self._connect_nonce, ack_nonce=self._ack_nonce
+        )
+        self._decoder = create_local_decoder(
+            local_key=self._local_key, connect_nonce=self._connect_nonce, ack_nonce=self._ack_nonce
+        )
         # Callback to decode messages and dispatch to subscribers
         self._data_received: Callable[[bytes], None] = decoder_callback(self._decoder, self._subscribers, _LOGGER)
 
+    async def _do_hello(self, local_protocol_version: LocalProtocolVersion) -> bool:
```

Comment:
It seems like it would be nice to remove side effects from this method. Can this be just about sending the hello, then the caller is response for converting the response to updated encoder params? Right now there are multiple methods with side effects (_do_hello, _update_encoder_decoder) and i would like to have fewer in the stack.

psuedocode but more something like this pattern:
```
async def hello()
    result = self.do_hello()
    self.update_encoder(params(result))
```

This gets easier to do given my suggestions above (args to update encoder) and below (reducing number of cases of calling _do_hello to 1) so make that more possible.